src/sql/workbench/api/node/extHostModelViewDialog.ts

```diff
@@ -95,12 +96,22 @@ class DialogImpl extends ModelViewPanelImpl implements sqlops.window.modelviewdi
 	public customButtons: sqlops.window.modelviewdialog.Button[];
 	private _message: sqlops.window.modelviewdialog.DialogMessage;
 	private _closeValidator: () => boolean | Thenable<boolean>;
+	private _operationHandler: BackgroundOperationHandler;
 
 	constructor(extHostModelViewDialog: ExtHostModelViewDialog,
-		extHostModelView: ExtHostModelViewShape) {
+		extHostModelView: ExtHostModelViewShape,
+		extHostTaskManagement: ExtHostBackgroundTaskManagementShape) {
 		super('modelViewDialog', extHostModelViewDialog, extHostModelView);
 		this.okButton = this._extHostModelViewDialog.createButton(DONE_LABEL);
 		this.cancelButton = this._extHostModelViewDialog.createButton(CANCEL_LABEL);
+		this._operationHandler = new BackgroundOperationHandler('dialog', extHostTaskManagement);
```

Comment:
What’s the ‘dialog’ argument for? Does it need to be unique if there are multiple dialogs?

Reply:
a guid will be added to this name. 

---
In reply to: [202498270](https://github.com/Microsoft/sqlopsstudio/pull/1929#discussion_r202498270) [](ancestors = 202498270)